### PR TITLE
(CDAP-4039) Added better defaults handling in FileBatchConfig. Also m…

### PR DESCRIPTION
**Pull request by bdmogal**
…ade FileBatchConfig parameters public so they can be accessed by derived class objects.

Build: http://builds.cask.co/browse/CDAP-DUT3003-5
Jira: [CDAP-4039](https://issues.cask.co/browse/CDAP-4039)

### CODE REVIEW

File: cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataTestBase.java [hunk 1]
```diff
@@ -166,6 +166,8 @@ protected void getPropertiesFromInvalidEntity(Id.Application app) throws Excepti
       getProperties(app);
       Assert.fail("Expected not to be able to get properties from invalid entity: " + app);
     } catch (NotFoundException expected) {
+      // expected
+      // expected
```
Comment (sagarkapare) [inline cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataTestBase.java]: twice?

**Comment by sagarkapare** [comment]
one minor comment, otherwise LGTM :+1:

File: cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataTestBase.java (hunk 1 above)
Comment (awholegunch) [inline cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataTestBase.java]: It was really, really expected. :)

**Comment by awholegunch** [comment]
I've made changes, and fixed the merge. Please review. Otherwise, LGTM.

**Comment by awholegunch** [comment]
[Docs Quick Build](http://builds.cask.co/browse/CDAP-DOB22-2)

**Comment by bdmogal** [comment]
Addressed comment, squashed commits. Merging.

**Pull request merged**